README.md

```diff
@@ -61,6 +61,43 @@ PyPI, which is recommended to restrict the access the action has.
 The secret used in `${{ secrets.PYPI_API_TOKEN }}` needs to be created on the
 settings page of your project on GitHub. See [Creating & using secrets].
 
+### Publishing with OpenID Connect
+
+**IMPORTANT**: This functionality is in beta, and will not work for you
+unless you're a member of the PyPI OIDC beta testers' group. For more
+information, see [warehouse#12965].
+
+This action supports PyPI's [OpenID Connect publishing]
+implementation, which allows authentication to PyPI without a manually
+configured API token or username/password combination. To perform
+[OIDC publishing][OpenID Connect Publishing] with this action, your project's
+OIDC publisher must already be configured on PyPI.
+
+To enter the OIDC flow, configure this action's job with the `id-token: write`
+permission and **without** an explicit username or password:
+
+```yaml
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for OIDC publishing
+    steps:
+    # retrieve your distributions here
```

Comment:
Note to myself: a part of me wants to showcase using something like [`step-security/harden-runner@v2`](https://github.com/python-attrs/attrs/blob/92f5eec/.github/workflows/pypi-package.yml#L77-L82) here, but I understand that this would make the example overloaded and would distract the readers. This probably deserves its own README section with a few snippets...